### PR TITLE
FEAT: Added usecontext files in context folder; Added react-router-dom

### PR DESCRIPTION
**Pull request by krishnaagarwal1506**
Main Changes in 5 Files
1 change in App due to parent branch changes
added react-router-dom package
added context folder with the usecontext hook files for Alert and Authentication

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
 
 const App = () => {
-  return <Home />;
+  return (
+    <Router>
+      <AlertProvider>
+        <Routes>
+          <Route>
+            <Route exact path="/" element={<Home />} />
+          </Route>
```
Comment (shelly19gkmit) [inline src/App.jsx:15]: why 2 <route>?

Comment (krishnaagarwal1506) [reply src/App.jsx:15]: Changed it.

File: src/context/AlertContext.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import React, { createContext, useState } from "react";
+
+export const AlertContext = createContext();
+
+const AlertContextProvider = ({ children }) => {
+  const [popup, setPopup] = useState({
```
Comment (shelly19gkmit) [inline src/context/AlertContext.jsx:6]: If this is a alert component, why have you defined state as popup?

File: src/context/AlertContext.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      value={{
+        popup,
+        showPopup,
+        hidePopup,
+        showAlertMessage,
+        isLoaderVisible,
+        showLoader,
+        hideLoader,
+      }}>
+      {children}
```
Comment (shelly19gkmit) [inline src/context/AlertContext.jsx:59]: Where is the cross icon and stylings?
Why are you not maintaing these here?

Comment (krishnaagarwal1506) [reply src/context/AlertContext.jsx:59]: these are the functionalities for toast and loader 
AlertContext provides me functionality of both 
the the components are not here 
they are in the component folder

File: src/context/AuthContext.jsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import React, { createContext, useState } from "react";
+
+export const AuthContext = createContext();
+
+const AuthProvider = (props) => {
```
Comment (shelly19gkmit) [inline src/context/AuthContext.jsx:5]: const AuthProvider = ({children}) => {

 <AuthContext.Provider value={{ isUserLoggedin, login, logout }}>
      {children}
    </AuthContext.Provider>

**Review by shelly19gkmit — APPROVED** [review]

File: src/context/AuthContext.jsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import React, { createContext, useState } from "react";
+
+export const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const [isUserLoggedin, setIsUserLoggedin] = useState(
+    sessionStorage.getItem("userEmail") ? true : false
+  );
+
```
Comment (shelly19gkmit) [inline src/context/AuthContext.jsx:9]: This is not a correct way of updating a state.
For reference - https://stackoverflow.com/questions/69101658/initial-state-update-using-usestate-is-not-working-in-reactjs

**Pull request merged**